Objective-C/Tests/URLEndpointListenerTest.m

```diff
@@ -437,4 +437,202 @@ - (void) testUnavailableNetworkInterface {
     [_listener stop];
 }
 
+- (void) testMultipleListenersOnSameDatabase {
+    Config* config = [[Config alloc] initWithDatabase: self.otherDB];
+    Listener* listener1 = [[Listener alloc] initWithConfig: config];
+    Listener* listener2 = [[Listener alloc] initWithConfig: config];
+    
+    NSError* err = nil;
+    Assert([listener1 startWithError: &err]);
+    AssertNil(err);
+    Assert([listener2 startWithError: &err]);
+    AssertNil(err);
+    
+    // replicate doc
+    [self generateDocumentWithID: @"doc-1"];
+    [self runWithTarget: listener1.localEndpoint
+                   type: kCBLReplicatorTypePushAndPull
+             continuous: NO
+          authenticator: nil
+             serverCert: (__bridge SecCertificateRef) listener1.config.tlsIdentity.certs[0]
+              errorCode: 0
+            errorDomain: nil];
+    
+    [listener1 stop];
+    [listener2 stop];
+    
+    AssertEqual(self.otherDB.count, 1);
```

Comment:
1. Can you add allow keychain check? 
2. Can you clean up the identities of the two listeners?